lib/ecto/parameterized_type.ex

```diff
@@ -99,6 +99,9 @@ defmodule Ecto.ParameterizedType do
   """
   @callback init(opts :: opts()) :: params()
 
+  @callback change(old_value :: any(), new_value :: any(), params :: params()) ::
+              {:ok, new_value :: any()} | {:error, Ecto.Changeset.error()} | :skip
+
```

Comment:
I believe we can define a default implementation for change that simply does:

    if equal?(old_value, new_value, params), do: :skip, else: {:ok, new_value}

Reply:
Also already done in: https://github.com/elixir-ecto/ecto/pull/3382/commits/dc286435cd22afaacd503763e05ef37f50038c01